packages/server/src/integrations/base/sql.ts

```diff
@@ -122,18 +122,22 @@ function generateSelectStatement(
     const fieldNames = field.split(/\./g)
     const tableName = fieldNames[0]
     const columnName = fieldNames[1]
-    if (
-      columnName &&
-      schema?.[columnName] &&
-      knex.client.config.client === SqlClient.POSTGRES
-    ) {
+    const columnSchema = schema?.[columnName]
+    if (columnSchema && knex.client.config.client === SqlClient.POSTGRES) {
       const externalType = schema[columnName].externalType
       if (externalType?.includes("money")) {
         return knex.raw(
           `"${tableName}"."${columnName}"::money::numeric as "${field}"`
         )
       }
     }
+    if (
+      knex.client.config.client === SqlClient.MS_SQL &&
+      columnSchema.type === FieldType.DATETIME &&
+      columnSchema.timeOnly
+    ) {
+      return knex.raw(`CONVERT(varchar, ${field}, 108) as "${field}"`)
```

Comment:
It would be worth commenting here that "108" is a "style code" that tells SQL Server to format as `hh:mm:ss`, to save the next person looking it up.